### PR TITLE
chore(ci): speed-up TravisCI builds [#141477825]

### PR DESCRIPTION
**Pull request by sbolel**
Reduces build time by ~175% (from  ~3.5 minutes to ~2 minutes)

### Summary of Changes

- update Node.js used by Travis to `v6.10.0` (Latest LTS)
- add caching `node_modules`
- move `npm i -g firebase-tools` to build/deploy.js (only install it when deploying)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,7 +26,6 @@ script:
   - npm run build:prod
 
 after_success:
-  - npm install -g firebase-tools
   - npm run deploy
   # - npm run codecov
```
Comment (prescottprue) [inline .travis.yml]: Travis actually added Firebase to the supported deploy stuff, so lets actually give this a try:

```yaml
deploy:
  skip_cleanup: true
  provider: firebase
  token:
    secure: $FIREBASE_TOKEN
    project: $TRAVIS_BRANCH
```

That will also mean we have to add this to `.firebaserc`:
```
"master": "devshare-int",
```

File: build/deploy.js [hunk 2]
```diff
@@ -24,20 +24,30 @@ const deployToFirebase = (cb) => {
     return
   }
 
+  debug('Installing firebase-tools...')
```
Comment (prescottprue) [inline build/deploy.js]: We can remove this file in favor of the Firebase travis deploy

File: build/deploy.js [hunk 3]
```diff
@@ -24,20 +24,30 @@ const deployToFirebase = (cb) => {
...
+
+  exec('npm install -g firebase-tools', (error, stdout) => {
```
Comment (prescottprue) [inline build/deploy.js]: If we are doing an async call to do this, we will need to do the two in succession, or we can't be sure that `firebase-tools` is installed before running the deploy script.

Is there a reason not to just keep the script right in the `travis.yml`

File: .travis.yml (hunk 1 above)
Comment (sbolel) [inline .travis.yml]: Nice, I'll make this change shortly!

File: build/deploy.js (hunk 3 above)
Comment (sbolel) [inline build/deploy.js]: My thinking was that firebase-tools does not have to be installed when not deploying. Currently, it installs even when the deploy script then doesn't use it. Should be a non-issue if we use Travis Firebase deploy.

Comment (prescottprue) [inline build/deploy.js]: Yeah, I see you there for sure. We just need to make sure the async flow is correct if doing it this way. Like you said though it should be a non-issue with Travis Firebase deploy.

Comment (sbolel) [inline build/deploy.js]: Oh i see what you mean haha. Its two execs in a row... Right...

**Comment by sbolel** [comment]
@prescottprue, updated Travis config based on your feedback. I'm not sure how to test it without changing the config, so let me know what you think.

File: .travis.yml [hunk 4]
```diff
@@ -25,9 +25,12 @@ script:
...
   - npm run build:prod
 
-after_success:
-  - npm run deploy
-  # - npm run codecov
+deploy:
+  skip_cleanup: true
+  provider: firebase
+  token:
+    project: $TRAVIS_BRANCH
```
Comment (prescottprue) [inline .travis.yml]: Project should be at the top level: https://docs.travis-ci.com/user/deployment/firebase/

File: build/deploy.js [hunk 5]
```diff
@@ -24,18 +24,6 @@ const deployToFirebase = (cb) => {
     return
   }
 
-  debug('Installing firebase-tools...')
```
Comment (prescottprue) [inline build/deploy.js]: Lets actually delete `deploy.js` and the npm script that calls it since it won't be used

File: .travis.yml (hunk 4 above)
Comment (sbolel) [inline .travis.yml]: Nice catch

File: build/deploy.js (hunk 5 above)
Comment (sbolel) [inline build/deploy.js]: That's what I figured! :P

**Comment by sbolel** [comment]
Updated to address feedback. Thanks!

**Comment by prescottprue** [comment]
Yes dude, absolutely perfect! Going to merge it for ya.

**Comment by sbolel** [comment]
Thanks @prescottprue!

**Pull request merged**